support/ruby/collins-state/lib/collins/state/mixin.rb

```diff
@@ -127,11 +127,11 @@ def state_specification asset
     updated = asset_from_cache asset
     result = updated.send(attribute_name.to_sym)
     if result then
-      res = JSON.parse(result) rescue nil
-      if res.nil? then # for backwards compatibility
-        res = JSON.parse(result, :create_additions => false)
+      res = JSON.parse(result, :create_additions => false) rescue nil
+      if not res.nil? then
         res = ::Collins::State::Specification.json_create(res) if (res.is_a?(Hash) and res.key?('data'))
```

Comment:
any reason not to combine the previous line's conditional with this one since there is only this one line in the if/then?